src/agents/_run_impl.py

```diff
@@ -51,6 +51,7 @@
     ToolOutputGuardrailTripwireTriggered,
     UserError,
 )
+from .extensions.handoff_filters import nest_handoff_history
```

Comment:
I understand other existing filters are in the extension file. but should this be under extensions? If we enable this by default (RunConfig#nest_handoff_history is true by default), the filter should be part of the main module. Also, importing extensions here looks like a code smell to me (meaning it may not be ideal in terms of pluggable module system design)